src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,186 @@
+
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() throws RegistrationException {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertDoesNotThrow(() -> registrationService.register(user));
+    }
+
+    @Test
+    void registerLoginExists_NotOk() throws RegistrationException {
+        User existingUser = new User();
+        existingUser.setLogin("existingUser");
+        existingUser.setPassword("password123");
+        existingUser.setAge(20);
+        storageDao.add(existingUser);
+
+        User newUserWithExistingLogin = new User();
+        newUserWithExistingLogin.setLogin("existingUser");
+        newUserWithExistingLogin.setPassword("newPassword123");
+        newUserWithExistingLogin.setAge(25);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUserWithExistingLogin);
+        });
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerPassword_NotOk() throws RegistrationException {
+        String[] invalidPasswords = {"1", "ab", "son", "user", "hell0"};
+        for (String password : invalidPasswords) {
+            user.setLogin("validLog");
+            user.setPassword(password);
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Password length must be at least 6 characters.", exception.getMessage());
+        }
+    }
+
+    @Test
+    void registerUserAge_NotOk() {
+        for (int i = 0; i < 18; i++) {
+            user.setLogin("validLogin");
+            user.setPassword("validPassword");
+            user.setAge(i);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Age must be at least 18 years old.", exception.getMessage());
```

Comment:
The loop in this test case might not be necessary as you can test the boundary condition (age just below 18) and the edge case (age exactly 18) without looping through all ages from 0 to 17. Consider simplifying the test to focus on the critical edge cases.